tests/integration/fetchMock.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+/**
+ * A `fetch` implementation for e2e testing.
+ *
+ * *Use only for testing purposes*.
+ *
+ * Example:
+ * ```
+ *     const fetchStub = sandbox.stub(global as any, "fetch");
+ *     fetchStub.callsFake(mockedFetch);
+ *     await someApi.loadSomethingWithFetch("https://example.com");
+ *     assert.equal(fetchStub, 1);
+ * ```
+ *
+ * @see [fetch documentation](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API)
+ */
+export async function fetchMock
+(
+  input: RequestInfo,
+  init?: RequestInit
+): Promise<Response> {
+  const url = typeof input === "object" ? input.url : input;
+  console.log("Requested url: " + url);
+  let result: string;
+  switch (url) {
+    case "https://api-lookup.data.api.platform.here.com/lookup/v1/resources/hrn:here:data:::test-hrn/apis/query/v1":
```

Comment:
This doesn't look very sustainable for all types of test cases?
I would rather go towards a custom test case expectation setup like gmock does:

```cpp
EXPECT_CALL(*network_mock_, Send(_, _, _, _, _))
      .WillOnce(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
                                       olp::http::HttpStatusCode::OK),
                                   kHttpResponseLookupQuery))
      .WillOnce(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
                                       olp::http::HttpStatusCode::OK),
                                   kHttpResponsePartition_269))
      .WillOnce(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
                                       olp::http::HttpStatusCode::OK),
                                   kHttpResponseLookupBlob))
      .WillOnce(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
                                       olp::http::HttpStatusCode::OK),
                                   kHttpResponseBlobData_269));
```

This way you can control from the test case itself what is expected and what is coming out including the number of expected requests so that you catch duplicate requests or unnecessary requests.

Reply:
Sure, thank you. We have plans to do something like that